frontend/src/app/shared/restaurants.service.ts

```diff
@@ -37,4 +38,8 @@ export class RestaurantsService {
     return this.googlePlacePhotoUrl + '?maxwidth=400&photoreference=' + photo + '&key=' + this.googleMapsKey;
   }
 
+  public incrementRestaurantCounter( restaurantId: string ): Observable<any> {
+    return this.httpClient.get(`/api/v1/increment/${restaurantId}`);
+  }
```

Comment:
you post data here 
/api/v1/increment ??

Reply:
/api/v1/restaurants/${restaurantId}/rewards
or something like this

Reply:
use Promise instead , it's simple and you don't have to unsubscribe

Reply:
Ok for the post method, but cant' use promise instead of observale httpClient's methods return observable